### PR TITLE
removed static create_method function pointer

### PR DESCRIPTION
**Pull request by sebastianbaginski**
Replaced with simple static method, defined in each `VidhanceProcessor` implementation file, this allows to clean up some code from headers.
Should work as long as you won't try to compile two or more VidhanceProcessor implementations at the same time.
@emilwestergren would this be better for https://github.com/vidhance/android-camera-wrapper/issues/31 ? It is a compile-time dependency if I understand correctly ?

### CODE REVIEW

File: nexus6/VidhanceProcessor.cpp [hunk 1]
```diff
@@ -8,13 +8,12 @@
...
+CameraWrapper* CameraWrapper::createMethod(const char* cameraId){
+    return new VidhanceProcessor(cameraId);
+}
 
 stride_t VidhanceProcessor::getStride(const sp<GraphicBuffer>& buffer) {
-	stride_t stride;
-	stride.vertical = ALIGN(buffer->height, 32);
-	stride.horizontal = ALIGN(buffer->width, buffer->usage & GraphicBuffer::USAGE_HW_VIDEO_ENCODER ? 64 : 32);
-	return stride;
+    return stride_t(ALIGN(buffer->width, buffer->usage & GraphicBuffer::USAGE_HW_VIDEO_ENCODER ? 64 : 32), ALIGN(buffer->height, 32));
```
Comment (sebastianbaginski) [inline nexus6/VidhanceProcessor.cpp]: contains the code from nexus6p as I couldn't compile and test it otherwise

File: HAL/CameraWrapper.h [hunk 2]
```diff
@@ -62,13 +58,14 @@ typedef struct {
 
 class CameraWrapper  {
 public:
+    static CameraWrapper* createMethod(const char* deviceId);
```
Comment (emilwestergren) [inline HAL/CameraWrapper.h]: Indentation is off.

File: HAL/CameraWrapperFactory.cpp [hunk 3]
```diff
@@ -65,7 +65,7 @@ int CameraWrapperFactory::camera_device_open(const struct hw_module_t *module, c
 int CameraWrapperFactory::cameraDeviceOpen(const char* camera_id, struct hw_device_t **hw_device)
 {
 	ALOGI("CameraWrapperFactory::cameraDeviceOpen");
-	CameraWrapper* wrapper = CameraWrapper::createMethod.createWrapper(camera_id);
+    CameraWrapper* wrapper = CameraWrapper::createMethod(camera_id);
```
Comment (emilwestergren) [inline HAL/CameraWrapperFactory.cpp]: Indentation

File: nexus5/VidhanceProcessor.cpp [hunk 4]
```diff
@@ -8,10 +8,9 @@
 #define LOG_TAG "VidhanceProcessor"
 
 namespace imint {
-CameraWrapper* VidhanceProcessor::createVidhanceProcessor(const char* cameraId) {
-	return new VidhanceProcessor(cameraId);
+CameraWrapper* CameraWrapper::createMethod(const char* cameraId){
+    return new VidhanceProcessor(cameraId);
```
Comment (emilwestergren) [inline nexus5/VidhanceProcessor.cpp]: Indentation

**Comment by emilwestergren** [comment]
You have indentation with spaces instead of tabs

**Comment by emilwestergren** [comment]
Looks like an improvement. What do you mean is compile-time dependent?

**Comment by sebastianbaginski** [comment]
Sorry must be default in my C++ editor (QtCreator)
> What do you mean is compile-time dependent?

I mean that previously this pointer was not meant to be reassigned at runtime.

**Comment by emilwestergren** [comment]
That's correct.

**Comment by sebastianbaginski** [comment]
Indentation is fixed now.

**Pull request merged**